protoc-gen-twirp_php/templates/service/AbstractClient.php

```diff
@@ -0,0 +1,245 @@
+<?php
+# Generated by the protocol buffer compiler (protoc-gen-twirp_php {{ .Version }}).  DO NOT EDIT!
+# source: {{ .File.Name }}
+
+declare(strict_types=1);
+
+namespace {{ .File | phpNamespace }};
+
+use Google\Protobuf\Internal\Message;
+use Http\Discovery\Psr17FactoryDiscovery;
+use Http\Discovery\Psr18ClientDiscovery;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use Twirp\Context;
+use Twirp\Error;
+use Twirp\ErrorCode;
+
+/**
+ * @internal {{ .Service | phpServiceName .File }}AbstractClient provides abstraction for JsonClient and Client (default).
```

Comment:
I'd probably expand on why this is internal (it's an implementation detail that's not covered by the BC promise)

Reply:
Did not understand your point about "bc promise".


This generated AbstractClient has func impl code, against rpc spec. And the JsonClient or Client, which are extenders, just implement abstract doRequest func. The client class were already final and still implements the same interface.

To avoid confusion, if convenient for you, can do 10m zoom call to sort it out. Or can continue here, fine with me.

Reply:
I meant that we don't guarantee that this abstract class will always be there as it is now or it will always work the same way. It's an internal detail and we need to make it clear as much as possible, that users MUST NOT use it.

The only thing that will/should remain backwards compatible is the two clients.

I'm okay with a call if the above doesn't explain better what I meant.

(From an other perspective: as a user I see the `@internal` tag, but I don't know why it's there or what to expect.)

Reply:
@sagikazarmark : Have made the suggested changes. Thanks.